display/ssd1306.h

```diff
@@ -225,15 +229,37 @@ class SSD1306 : public I2CDevice, Looper, StateMachine, SaberBase, private Audio
 	}
 	if (eof_) {
 	  // STDOUT << "EOF " << frame_count_ << "\n";
-	  screen_ = SCREEN_PLI;
-	  if (frame_count_ == 1) return 5000;
-	  return FillFrameBuffer();
+    if (!SaberBase::IsOn()) {
+	   screen_ = SCREEN_PLI;
+	    if (frame_count_ == 1) return font_config.ProffieOSImageDuration;
+	    return FillFrameBuffer();
+    }
 	}
 	frame_count_++;
-	if (looped_frames_ > 1 && millis() - loop_start_ > 5000) {
-	  STDOUT << "END OF LOOP\n";
-	  screen_ = SCREEN_PLI;
-	}
+  if (SaberBase::IsOn()) {
+    if (looped_frames_ == 1 && frame_count_ == 1) {
```

Comment:
I think I would like to change the structure a bit, and add some comments, something like:

    if (looped_frames == 1) {
        // Single frame
        if (frame_count == 1) return font_config.ProffieOSImageDuration;
        if (SaberBase::Lockup()) {
        } else {
        }
     } else {
        // Looped
     } 

There shouldn't be a need to check looped_frames in so many places.

Reply:
Should be a bit more simplified now.